apps/cowswap-frontend/src/legacy/state/swap/utils.ts

```diff
@@ -20,44 +24,69 @@ export function isWrappingTrade(
   )
 }
 
-export function logTradeDetails(trade: TradeGp | undefined, allowedSlippage: Percent) {
-  // don't do anything outside of dev env
-  if (!trade || process.env.NODE_ENV !== 'development') return
-
-  const exactIn = trade.tradeType === TradeType.EXACT_INPUT
-
-  // Log Exact In Trade info
-  if (exactIn) {
-    console.debug(
-      `[SwapMod::[SELL] Trade Constructed]`,
-      `
-      Type: SELL
-      ==========
-      Input Amount:         ${trade.inputAmount.toExact()}
-      Output Amount:        ${trade.outputAmount.toExact()}
-      ==========
-      Fee Amount [as SELL]: ${trade.fee?.feeAsCurrency?.toExact()} ${trade.inputAmount.currency.symbol}
-      Fee Amount [as BUY]:  ${
-        trade.outputAmountWithoutFee && trade.outputAmountWithoutFee.subtract(trade.outputAmount).toExact()
-      } ${trade.outputAmount.currency.symbol}
-      ==========
-      Minimum Received:     ${trade.minimumAmountOut(allowedSlippage).toExact()}
-    `
-    )
-  } else {
-    // Log Exact Out Trade info
-    console.debug(
-      `[SwapMod::[BUY] Trade Constructed]`,
-      `
-      Type: BUY
-      =========
-      Input Amount [w/FEE]: ${trade.inputAmountWithFee.toExact()}
-      Output Amount:        ${trade.outputAmount.toExact()}
-      =========
-      Fee Amount [as SELL]: ${trade.fee?.feeAsCurrency?.toExact()} ${trade.inputAmount.currency.symbol}
-      =========
-      Maximum Sold:         ${trade.fee?.feeAsCurrency && trade.maximumAmountIn(allowedSlippage).toExact()}
-    `
-    )
+function parseIndependentFieldURLParameter(urlParam: any): Field {
```

Comment:
Just moved functions from `hooks.tsx` using auto refactoring.

Reply:
To fix circular dependency problem

Reply:
What do you mean by `auto refactoring`?

Reply:
<img width="782" alt="image" src="https://github.com/cowprotocol/cowswap/assets/7122625/911816ac-c1ae-48bb-94ba-bc52bc2766a8">